suite.lisp

```diff
@@ -0,0 +1,77 @@
+(in-package #:org.shirakumo.trivial-benchmark)
+
+(defvar *benchmark-packages* (make-hash-table :test 'eq)
+  "A table mapping package objects (which are benchmark packages) to a list of fbound symbols.")
+
+(defun find-benchmark-package (package-designator)
+  "Find the benchmark package designated by PACKAGE-DESIGNATOR and return it, otherwise return NIL if it doesn't exist."
+  (let ((package-object (find-package package-designator)))
+    (if (or (null package-object)
+            (null (nth-value 1 (gethash package-object *benchmark-packages*))))
+        nil
+        package-object)))
+
+(defmacro define-benchmark-package (name &body package-options)
+  "Define a package as if by DEFPACKAGE, except that both the COMMON-LISP and TRIVIAL-BENCHMARK packages are used, and that this package is designated as one containing benchmarks."
+  `(progn
+     (defpackage ,name
+       (:use #:trivial-benchmark #:cl)
+       ,@package-options)
+     (setf (gethash (find-package ',name) *benchmark-packages*) nil)))
+
+(defvar *current-suite-report*)         ; Table mapping fn name -> metric alist
+(defvar *current-benchmark*)            ; Name of current benchmark
+(defvar *current-timer*)                ; Timer used in the current benchmark.
+
+(defmacro measure-forms (&body forms)
```

Comment:
Something that is closer in name to the original `with-sampling` would probably avoid confusion about the difference. How about `with-bench-sampling`?